test/index.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+var assert = require('assert');
+var intercept = require('intercept-stdout');
+
+var Z = require('sanctuary-type-classes');
+
+var createRunner = require('..');
+
+
+function eq(actual, expected) {
+  assert.strictEqual(arguments.length, eq.length);
+  assert.strictEqual(Z.toString(actual), Z.toString(expected));
+  assert.strictEqual(Z.equals(actual, expected), true);
+}
+
+function expectOutput(expectations) {
+  var idx = 0;
+  var unhook = intercept(function(text) {
+    text.split('\n').forEach(function(line) {
+      if (idx >= expectations.length) {
+        unhook();
+        throw new Error(
+          'More lines than expected were printed:\n\n' +
+          JSON.stringify(line)
+        );
+      }
+      if (expectations[idx].test(line) === false) {
+        unhook();
+        throw new Error(
+          'A printed line did not meet expectations:\n\n' +
+          JSON.stringify(line) + '\n\n' +
+          'It does not match expression:\n\n' +
+          expectations[idx].toString()
+        );
+      }
+      idx += 1;
+    });
+  });
+  return function() {
+    unhook();
+    if (idx < expectations.length - 1) {
+      throw new Error(
+        'Expected ' +
+        String(idx - expectations.length) +
+        ' more lines to be printed.'
+      );
+    }
+  };
+}
+
+var mocklibA = {lib: 'a'};
+var mocklibB = {lib: 'b'};
+
+test('createRunner', function() {
+  eq(typeof createRunner, 'function');
+  eq(createRunner.length, 3);
+  eq(typeof createRunner(mocklibA, mocklibB, {}), 'function');
+  eq(createRunner(mocklibA, mocklibB, {}).length, 1);
+});
+
+function specUniform(lib) {
+  eq(lib.lib === 'a' || lib.lib === 'b', true);
+}
+
+var specSplit = {
+  left: function(lib) { eq(lib.lib, 'a'); },
+  right: function(lib) { eq(lib.lib, 'b'); }
+};
+
+var runnerUniform = createRunner(mocklibA, mocklibB, {
+  'mock/first': specUniform,
+  'mock/second': specUniform
+});
+
+var runnerSplit = createRunner(mocklibA, mocklibB, {
+  'mock/first': specSplit,
+  'mock/second': specSplit
+});
+
+function runAndAssert(options, expectations) {
+  var doneUniform = expectOutput(expectations);
+  runnerUniform(options);
+  doneUniform();
+  var doneSplit = expectOutput(expectations);
+  runnerSplit(options);
+  doneSplit();
+}
+
+test('runBenchmarks', function() {
+  runAndAssert({
+    colors: false,
+    leftHeader: 'custom left',
+    match: '*/first',
+    minSamples: 1,
+    rightHeader: 'custom right',
+    significantChange: 0
+  }, [
+    /^# 1\/1: mock\/first\r/,
+    /^┌[─┬]+┐$/,
+    /^│ suite +│ custom left +│ custom right +│ change +│ α │$/,
+    /^├[─┼]+┤$/,
+    /^│ mock\/first +│(?: [0-9,]+ Hz ±\d+\.\d+% \(n \d+\) +│){2} [+-]\d+\.\d% +│ [ ✓✗] │$/,
```

Comment:
We could use `[\d,]` in place of `[0-9,]` (or use `[0-9]` in place of `\d`). Also, I find `[/]` and `[.]` slightly nicer to read than `\/` and `\.`.